app/services/usps_in_person_proofing/proofer.rb

```diff
@@ -153,6 +153,9 @@ def faraday
     # already cached.
     # @return [Hash] Headers to add to USPS requests
     def dynamic_headers
+      if Rails.cache.redis.ttl(AUTH_TOKEN_CACHE_KEY) != -2 && Rails.cache.redis.ttl(AUTH_TOKEN_CACHE_KEY) <= 0.5 # rubocop:disable Layout/LineLength
+        retrieve_token!
```

Comment:
Can we pull the `ttl` once? Just in case there's some weird condition that changes between calls?

Also can we move the `0.5` magic number into a constant?

```suggestion
      key_ttl = Rails.cache.redis.ttl(AUTH_TOKEN_CACHE_KEY)
      if key_ttl != -2 && key_ttl <= 0.5
        retrieve_token!
```

Reply:
Good idea. 👍🏻

Reply:
Also what do you think about bumping the `0.5` seconds to maybe `5` seconds?

Reply:
@NavaTim Why `5` seconds?  I'm fine with increasing the buffer, but curious about why you chose that number.

Reply:
I'm specifically thinking here about an amount of time approximating the upper end of what it'd take to serve an average request (maybe 75th percentile or higher) - including backend calls to DB, cache, and a potentially slow USPS API. Part of the goal is to prevent requests from failing midway through because the token wasn't updated early enough. We would also need to add a little buffer for the token update process itself.

Reply:
Makes sense, thank you. 🙏🏻